src/icp_v2.cc

```diff
@@ -680,16 +662,26 @@ icpHandleUdp(int sock, void *)
         debugs(12, 4, "icpHandleUdp: FD " << sock << ": received " <<
                (unsigned long int)len << " bytes from " << from);
 
-#ifdef ICP_PACKET_DUMP
-
-        icpPktDump(buf);
-#endif
-
         if ((size_t) len < sizeof(icp_common_t)) {
             debugs(12, 4, "icpHandleUdp: Ignoring too-small UDP packet");
             break;
         }
 
+        auto pkt = reinterpret_cast<icp_common_t *>(buf);
+        struct in_addr a;
+        a.s_addr = pkt->shostid;
+        debugs(12, 2, "ICP Client remote=" << from << " FD " << sock);
+        debugs(12, 2, "ICP Client REQUEST:\n---------\n" <<
+               "opcode=(" << std::setw(3) << pkt->opcode << ")" << icp_opcode_str[pkt->opcode] <<
+               ", version=" << pkt->version <<
+               ", length=" << ntohs(pkt->length) <<
+               ", reqnum=" << std::setw(8) << ntohl(pkt->reqnum) <<
+               ", flags=(" << std::hex << std::setw(8) << ntohl(pkt->flags) << ")" <<
+               ", shostid=" << Ip::Address(a) <<
+               Debug::Extra <<
+               Raw(nullptr, (buf+sizeof(icp_common_t)), ntohs(pkt->length)).minLevel(9) <<
```

Comment:
Consider dropping the minLevel() part -- IIRC, it will be added automatically for large pkt-length values.

Reply:
I considered it. But to keep the log output consistent and because even short URI which most of these will be may contain credentials I am putting level-9 along with all other full-data dumps.

Reply:
Not sure what inconsistencies you perceive.

If we are concerned about credentials, then we should make that concern explicit IMO. If you are not around, somebody else can easily change or remove that debugging level without realizing that the information is sensitive. Not all level-9 information is sensitive, of course! We can address this by adding and using Raw::sensitive() or a similar method (which can just call minLevel(9) today, I guess, but future implementations could be different).

Reply:
I am trying to make this output as similar as possible to the level-2 output already used by HTTP and FTP. Those displays are header only, without any payload logged until higher levels (ie DBG_DATA / 9).

Also, I think it a bad idea to create any distinction between small vs large payloads. "KISS" principle, etc. if not to avoid issues from mistakes in admin expectations.